src/tabs-motion/types.js

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright (c) 2018-2020 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+// @flow
+
+import * as React from 'react';
+import {
+  ORIENTATION,
+  FILL,
+  KEYBOARD_ACTIVATION,
+  STATE_CHANGE_TYPE,
+} from './constants.js';
+
+import type {OverrideT} from '../helpers/overrides.js';
+import type {IconPropsT} from '../icon/types.js';
+
+export type OrientationT = $Values<typeof ORIENTATION>;
+export type FillT = $Values<typeof FILL>;
+export type KeyboardActivationT = $Values<typeof KEYBOARD_ACTIVATION>;
+export type StateChangeT = $Values<typeof STATE_CHANGE_TYPE>;
+
+export type TabsOverridesT = {|
+  Root?: OverrideT,
+  TabList?: OverrideT,
+  TabHighlight?: OverrideT,
+  TabBorder?: OverrideT,
+|};
+
+export type TabOverridesT = {|
+  Tab?: OverrideT,
+  ArtworkContainer?: OverrideT,
+  TabPanel?: OverrideT,
+|};
+
+export type StatefulTabsStateT = {|
+  activeKey: React.Key,
```

Comment:
Interesting use of `React.Key`. Is it just convenient to use this versus `string | number`? Not sure if it's worth worrying about them changing the type offsetting the cost of a bit of code-reuse

Reply:
If React ever updated `key` I'd actually *prefer* to update the Tabs component to still use that. I really like that the currently selected Tab is based on the React key. It feels like a nice way to leverage a pattern folks should already be familiar with, if that makes any sense. I imagine if they ever made `key` more complicated they'd also have a way of validating equality.